assets/js/util/report-validation.js

```diff
@@ -99,6 +104,52 @@ export function isValidOrders( orders ) {
 	return false;
 }
 
+/**
+ * Verifies that order definitions are valid for GA4. It should be an array
+ * of objects where each object has either a "metric" or a "dimension" property,
+ * and an optional "desc" property. The "metric" and "dimension" properties should
+ * be objects with "metricName" and "dimensionName" properties respectively.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object[]} orders The order definitions to check.
+ * @return {boolean} TRUE if order definitions are valid, otherwise FALSE.
+ */
+export function isValidOrdersGA4( orders ) {
+	if ( ! Array.isArray( orders ) ) {
+		return false;
+	}
+
+	return orders.every( ( order ) => {
+		if ( ! isPlainObject( order ) ) {
+			return false;
+		}
+
+		if (
+			order.hasOwnProperty( 'desc' ) &&
+			typeof order.desc !== 'boolean'
+		) {
+			return false;
+		}
+
+		if ( order.metric ) {
+			if ( order.dimension ) {
+				return false;
+			}
+			return typeof order.metric?.metricName === 'string';
+		}
+
+		if ( order.dimension ) {
+			if ( order.metric ) {
+				return false;
+			}
```

Comment:
This check is dead code because the `metric` property is already processed in the previous condition. We can remove it.

Reply:
Good point, thanks! Removed :+1: